ddtrace/propagation/http.py

```diff
@@ -950,6 +966,18 @@ def parent_call():
             for key in span_context._baggage:
                 headers[_HTTP_BAGGAGE_PREFIX + key] = span_context._baggage[key]
 
+        if hasattr(ddtrace, "tracer") and hasattr(ddtrace.tracer, "sample"):
+            if non_active_span is not None:
```

Comment:
Can we simplify this logic to: 

- If tracer.current_context == span_context THEN call tracer.current_root_span().sample() 
- If tracer.current_context != span_context then call non_active.sample() 
   -  if non_active is None then log an actionable warning

Reply:
This way users will be notified right away if httpinject is not sampling some correcty

Reply:
I changed the code a bit to rely on span._local_root, which simplifies the logic well I think.